packages/core/util/layouts/GranularRectLayout.ts

```diff
@@ -149,17 +151,13 @@ class LayoutRow<T> {
     if (left < this.rowState.offset) {
       // expand to new left - the whole current length (or 0)
       // additionalLength = (offset - left) + currLength = -(left - offset) + currLength
-      const additionalLength = Math.max(
```

Comment:
I think I may have originally meant to put `Math.min` here instead of `Math.max`. Basically the idea was to keep the offset from needlessly going lower than 0.

Reply:
hmm, well does it cause any trouble to not have it? math.max/min

Reply:
Mathematically it doesn't matter, but fixing it to be `Math.min` instead of getting rid of it will help keep the array size down, which could help memory usage. (Also, I found negative offsets to be a bit hard to reason about, but it's not that often that you actually have to think about that.)

Reply:
ok, I flipped it to Math.min now